packages/ui/src/components/TextInput/index.tsx

```diff
@@ -551,6 +544,11 @@ export const TextInput = forwardRef<
       dataTestId,
     ])
 
+    const rightElementPadding =
+      rightComponentsArray.length * 32 +
+      (required ? 32 : 0) +
+      (unit || hasRightElement ? 8 : 0)
```

Comment:
I struggle to understand why 8 here mb we can explain the calculation somewhere
+ since the spacing is now in the theme, mb we should just calculate space unit here, then multiply by theme.space['1'] using a calc-width ?

It works for everycase, but the last one isn't perfect, the padding is higher than the other ones

![Capture d’écran 2023-06-06 à 12 35 49](https://github.com/scaleway/scaleway-ui/assets/8116991/8c212340-6d73-45bf-ab89-a0cfc5eaecdb)

Reply:
As discussed, I've refactored the way it works by using a factor and multiplying it by a theme value.